proposals/lip-0045.md

```diff
@@ -1549,7 +1549,7 @@ On the mainchain, the following checks are performed:
   * check that `totalWeight//3 + 1 <= certificateThreshold <= totalWeight`, where `//` indicates integer division;
   * check that the corresponding `validatorsHash` stored in `chainInfo.chainData.lastCertificate.validatorsHash` matches with the value computed from `activeValidators` and `certificateThreshold`.
 * For each entry `chainInfo` in `chainInfos`, `chainInfo.chainData.status == CHAIN_STATUS_TERMINATED` if and only if a corresponding entry (i.e., with `chainID == chainInfo.chainID`) exists in `terminatedStateAccounts`.
-* Each entry `stateAccount` in `terminatedStateAccounts` has a unique `stateAccount.chainID` and `terminatedStateAccounts` is ordered lexicographically by `stateAccount.chainID`.
+* Each entry `stateAccount` in `terminatedStateAccounts` has a unique `stateAccount.chainID` and `terminatedStateAccounts` is ordered lexicographically by `stateAccount.chainID`. Furthermore for each entry it holds `stateAccount.chainID != ownChainAccount.chainID`.
```

Comment:
I would check this as we do above:

* `stateAccount.chainID != getMainchainID()`;
* `stateAccount.chainId[0] == getMainchainID()[0]`.

Reply:
Ok makes sense. I guess also add this check for first byte in the corresponding sidechain check (line 1586)?

Reply:
Added.